cedar-policy-core/src/est/expr.rs

```diff
@@ -1011,25 +1011,51 @@ impl TryFrom<&Node<Option<cst::Unary>>> for Expr {
     type Error = ParseErrors;
     fn try_from(u: &Node<Option<cst::Unary>>) -> Result<Expr, ParseErrors> {
         let u_node = u.ok_or_missing()?;
-        let inner = (&u_node.item).try_into()?;
+
         match u_node.op {
-            Some(cst::NegOp::Bang(0)) => Ok(inner),
-            Some(cst::NegOp::Bang(1)) => Ok(Expr::not(inner)),
+            Some(cst::NegOp::Bang(0)) => Ok((&u_node.item).try_into()?),
+            Some(cst::NegOp::Bang(1)) => Ok(Expr::not((&u_node.item).try_into()?)),
             Some(cst::NegOp::Bang(2)) => {
                 // not safe to collapse !! to nothing
-                Ok(Expr::not(Expr::not(inner)))
+                Ok(Expr::not(Expr::not((&u_node.item).try_into()?)))
             }
             Some(cst::NegOp::Bang(n)) => {
                 if n % 2 == 0 {
                     // safe to collapse to !! but not to nothing
-                    Ok(Expr::not(Expr::not(inner)))
+                    Ok(Expr::not(Expr::not((&u_node.item).try_into()?)))
                 } else {
                     // safe to collapse to !
-                    Ok(Expr::not(inner))
+                    Ok(Expr::not((&u_node.item).try_into()?))
                 }
             }
-            Some(cst::NegOp::Dash(0)) => Ok(inner),
+            Some(cst::NegOp::Dash(0)) => Ok((&u_node.item).try_into()?),
             Some(cst::NegOp::Dash(mut num_dashes)) => {
+                let inner = match &u_node.item {
+                    Node {
+                        node:
+                            Some(cst::Member {
+                                item:
+                                    Node {
+                                        node:
+                                            Some(cst::Primary::Literal(Node {
+                                                node: Some(cst::Literal::Num(x)),
+                                                ..
+                                            })),
+                                        ..
+                                    },
+                                access,
+                            }),
+                        ..
+                    } if access.is_empty()
+                        && (x
+                            .checked_sub(1)
+                            .map(|y| y == InputInteger::MAX as u64)
+                            .unwrap_or(false)) =>
+                    {
+                        Expr::ExprNoExt(ExprNoExt::Value(CedarValueJson::Long(InputInteger::MIN)))
+                    }
```

Comment:
don't we need to subtract one from `num_dashes` in this case?  (and do we have a test for this code, that makes sure the value is correct, or preserved across various roundtrips?)